core/api/src/test/java/com/microsoft/gctoolkit/test/time/DateTimeStampTest.java

```diff
@@ -353,5 +353,46 @@ void timeSpanInMinutes() {
         diff = a.timeSpanInMinutes(b);
         assertEquals((.18 - .19)/60d, diff, .001);
     }
+    @Test
+    void testCompareLessThan(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T09:10:00.586-0100");
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.587-0100");
+        assertEquals(-1, smaller.compareTo(greater));
+    }
+    @Test
+    void testCompareGreaterThan(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T09:10:00.586-0100");
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.587-0100");
+        assertEquals(1, greater.compareTo(smaller));
+    }
+    @Test
+    void testCompareEquals(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T10:10:00.586-0100");
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.586-0100");
+        assertEquals(0, greater.compareTo(smaller));
+    }
+    @Test
+    void testCompareEqualsTimeStamp(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T10:10:00.586-0100",123);
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.586-0100",123);
+        assertEquals(0, greater.compareTo(smaller));
+    }
+    @Test
+    void testCompareGreaterTimeStamp(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T10:10:00.586-0100",122);
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.586-0100",123);
+        assertEquals(1, greater.compareTo(smaller));
+    }
+    @Test
+    void testCompareSmallerTimeStamp(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T10:10:00.586-0100",123);
+        DateTimeStamp greater = new DateTimeStamp("2018-04-04T10:10:00.586-0100",124);
+        assertEquals(-1, smaller.compareTo(greater));
+    }
+    @Test
+    void testCompareNullValue(){
+        DateTimeStamp smaller = new DateTimeStamp("2018-04-04T09:10:00.586-0100");
+        assertEquals(-1, smaller.compareTo(null));
+    }
```

Comment:
Seems, needs improvement in code when we have 
`new DateTimeStamp(123).compareTo(new `DateTimeStamp("2018-04-04T10:10:00.586-0100"))`
Should I assume when DateTime is null then directly compare timestamp, instead of comparing DateTime first?

Reply:
This is where I'd like @kirk-microsoft to chime in. KIrk, should the compare just be timeStamp only.

Reply:
if there is a time stamp, then I think we can restrict the comparison to that. At the moment, we always generate a synthetic time stamp but I think that it would be prudent to check the date stamp in the case where the time stamp is missing.